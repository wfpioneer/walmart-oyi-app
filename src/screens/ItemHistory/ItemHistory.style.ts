import { StyleSheet } from 'react-native';
import COLOR from '../../themes/Color';

const styles = StyleSheet.create({
  historyCard: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    backgroundColor: COLOR.WHITE,
    paddingHorizontal: 10,
    paddingVertical: 5,
    marginBottom: 2
  }
});

export default styles;
