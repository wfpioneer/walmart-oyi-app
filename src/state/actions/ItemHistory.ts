import { ItemHistoryI } from '../../models/ItemDetails';

export const SET_ITEM_HISTORY = 'ITEM_HISTORY/SET_HISTORY';
export const CLEAR_ITEM_HISTORY = 'ITEM_HISTORY/CLEAR_ITEM_HISTORY';

export const setItemHistory = (data: ItemHistoryI[], title: string) => ({
  type: SET_ITEM_HISTORY,
  payload: { data, title }
} as const);

export const clearHistory = () => ({
  type: CLEAR_ITEM_HISTORY
} as const);

export type Actions =
  | ReturnType<typeof setItemHistory>
  | ReturnType<typeof clearHistory>
