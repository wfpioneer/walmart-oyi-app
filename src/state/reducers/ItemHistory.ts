import { ItemHistoryI } from '../../models/ItemDetails';
import {
  Actions,
  CLEAR_ITEM_HISTORY,
  SET_PICK_HISTORY
} from '../actions/ItemHistory';

export interface ItemHistoryState {
   data: ItemHistoryI[];
   title: string
  }

export const initialState: ItemHistoryState = {
  data: [],
  title: ''
};

export const ItemHistory = (state = initialState, action: Actions) => {
  switch (action.type) {
    case SET_PICK_HISTORY:
      return {
        ...state,
        data: action.payload.map(itm => ({
          id: itm.id,
          date: itm.createTS,
          qty: itm.itemQty
        })),
        title: 'ITEM.PICK_HISTORY'
      };
    case CLEAR_ITEM_HISTORY:
      return {
        ...state
      };
    default:
      return state;
  }
};
