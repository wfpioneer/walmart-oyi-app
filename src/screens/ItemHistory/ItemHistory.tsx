import moment from 'moment';
import React from 'react';
import { FlatList, Text, View } from 'react-native';
import { useTypedSelector } from '../../state/reducers/RootReducer';
import { ItemHistoryState } from '../../state/reducers/ItemHistory';
import styles from './ItemHistory.style';

const ItemHistory = () => {
  const { data } = useTypedSelector(state => state.ItemHistory) as ItemHistoryState;
  const today = new Date();
  const dateBefSixMonth = new Date(today.getTime() - (180 * 24 * 60 * 60 * 1000));
  const sortedData = data.filter(item => new Date(item.date) > dateBefSixMonth)
    .sort((a, b) => {
      const date1 = new Date(a.date);
      const date2 = new Date(b.date);
      return date2 > date1 ? 1 : -1;
    });
  return (
    <View>
      <FlatList
        data={sortedData}
        keyExtractor={item => item.id.toString()}
        renderItem={({ item }) => (
          <View style={styles.historyCard}>
            <Text>{moment(item.date).format('YYYY-MM-DD')}</Text>
            <Text>{item.qty}</Text>
          </View>
        )}
      />
    </View>
  );
};

export default ItemHistory;
