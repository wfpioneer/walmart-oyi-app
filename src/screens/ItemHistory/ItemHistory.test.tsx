import React from 'react';
import { render } from '@testing-library/react-native';
import { Provider } from 'react-redux';
import ItemHistory from './ItemHistory';
import store from '../../state/index';

describe('Tests Rendering', () => {
  it('Renders item history flat list', () => {
    const component = (
      <Provider store={store}>
        <ItemHistory />
      </Provider>
    );
    const { toJSON } = render(component);
    expect(toJSON()).toMatchSnapshot();
  });
});
