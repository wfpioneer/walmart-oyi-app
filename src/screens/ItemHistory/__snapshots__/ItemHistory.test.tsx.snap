// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests Rendering Renders item history flat list 1`] = `
<View>
  <RCTScrollView
    data={Array []}
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    viewabilityConfigCallbackPairs={Array []}
  >
    <View />
  </RCTScrollView>
</View>
`;
